plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -206,7 +217,8 @@ public static IRuntime createAppEngineFacetRuntime(IProgressMonitor monitor)
   /**
    * Installs Java 1.7 facet if it doesn't already exist in <code>factedProject</code>
    */
-  private static void installJavaFacet(IFacetedProject facetedProject, IProgressMonitor monitor)
+  private static void addJavaFacetToBatchQueue(IFacetedProject facetedProject,
+      Set<IFacetedProject.Action> batchQueue)
```

Comment:
Good catch! It's a set. (WTP will determine the order of actions according to dependencies.)